centreon/www/class/CentreonDbConfig.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * Copyright 2005 - 2024 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+/**
+ * Class CentreonDbConfig
+ *
+ * @class CentreonDbConfig
+ */
+final class CentreonDbConfig
+{
+
+    /**
+     * @param string $dbHost
+     * @param string $dbUser
+     * @param string $dbPassword
+     * @param string $dbName
+     * @param int $dbPort
+     * @param string $dbDriver
+     */
+    public function __construct(
+        public readonly string $dbHost,
+        public readonly string $dbUser,
+        public readonly string $dbPassword,
+        public readonly string $dbName,
+        public readonly int $dbPort = 3306,
+        public readonly string $dbDriver = 'mysql'
+    ) {
+    }
+
+    /**
+     * @return string
+     */
+    public function getPdoDsn(): string
+    {
+        return sprintf(
+            "%s:dbname=%s;host=%s;port=%s",
+            $this->dbDriver,
+            $this->dbName,
+            $this->dbHost,
+            $this->dbPort
+        );
+    }
```

Comment:
The idea to provide the dbDriver and make the CentreonDbConfig database agnostic was cool, but your class is still very MySQL-ish: the default port is MySQL's and the DSN string format won't work with other SGBD.

`"%s:dbname=%s;host=%s;port=%s"` only works for Mysql/MariaDB
The pattern is different if you decide to use postgres, sqlserver, oracle etc.

I'm not sure I would bother trying to please this level of abstraction for now

Reply:
fixed